diff --git a/flake.nix b/flake.nix
--- a/flake.nix
+++ b/flake.nix
@@ -216,7 +216,7 @@
                   set -euo pipefail
                   if [ ! -s '$host_key_path' ]; then
                     rm -f '$host_key_path' '$host_pubkey_path'
-                    ssh-keygen -t ed25519 -N '' -f '$host_key_path'
+                    ssh-keygen -t ed25519 -N "" -f '$host_key_path'
                   fi
                   if [ ! -s '$host_pubkey_path' ]; then
                     ssh-keygen -y -f '$host_key_path' > '$host_pubkey_path'
